src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -29,34 +43,44 @@ export const MyProfile = () => {
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const user = useSelector((store: RootState) => store.user)!; // If user has been empty this route would already be blocked
 
-  const personId = useSelector((store: RootState) => store.user?.cristinId) ?? '';
+  const personId = user?.cristinId ?? '';
 
   const personQuery = useQuery({
     queryKey: [personId],
     queryFn: () => fetchPerson(personId),
     onError: () => dispatch(setNotification({ message: t('feedback.error.get_person'), variant: 'error' })),
   });
-  const person = personQuery.data;
 
-  const hasActiveEmployment = filterActiveAffiliations(person?.affiliations).length > 0;
+  const person = personQuery.data;
 
   const firstName = getValueByKey('FirstName', person?.names);
   const lastName = getValueByKey('LastName', person?.names);
   const personPreferredFirstName = getValueByKey('PreferredFirstName', person?.names);
   const personPreferredLastName = getValueByKey('PreferredLastName', person?.names);
   const [editPreferredNames, setEditPreferredNames] = useState(false);
-  const personTelephone = person?.contactDetails?.telephone;
 
   const initialValues: CristinPersonFormData = {
     preferredFirstName: personPreferredFirstName ? personPreferredFirstName : firstName,
     preferredLastName: personPreferredLastName ? personPreferredLastName : lastName,
+    contactDetails: {
+      telephone: person?.contactDetails?.telephone ?? '',
+      email: person?.contactDetails?.email ?? '',
+      webPage: person?.contactDetails?.webPage ?? '',
+    },
   };
 
   const updatePerson = async (values: CristinPersonFormData) => {
     if (user.cristinId) {
+      const trimValue = (value: string | undefined | null) => (value === '' ? null : value?.trim());
+
       const payload: CristinPersonFormData = {
-        preferredFirstName: values.preferredFirstName === '' ? null : values.preferredFirstName?.trim(),
-        preferredLastName: values.preferredLastName === '' ? null : values.preferredLastName?.trim(),
+        preferredFirstName: trimValue(values.preferredFirstName),
+        preferredLastName: trimValue(values.preferredLastName),
+        contactDetails: {
+          telephone: trimValue(values.contactDetails?.telephone),
+          email: trimValue(values.contactDetails?.email),
+          webPage: trimValue(values.contactDetails?.webPage),
```

Comment:
nå er det veldig flisespikkeri fra meg da, men det bare føles litt bortkastet å lage en funksjon `trimValue` som gjør dette, så lenge den bare kaller en native funksjon selv også (`trim()`). Kunne løst det på samme måte som du har gjort i intialValues:
```suggestion
          webPage: values.contactDetails?.webPage?.trim() || null,
```
Antar dette ble gjort for at verdi skal bli satt til `null` når det er tom streng

Reply:
for reference:
![bilde](https://github.com/BIBSYSDEV/NVA-Frontend/assets/6313468/44913752-0392-4b7f-8326-3435cf9eef9e)

`??` tar alt `null` og `undefined`, mens `||` tar alt som er falsy, inkludert `""` og `0`